### DISCUSSION TITLE
[ROFI-SCRIPT] $ROFI_INFO for custom entries.

### DISCUSSION THREAD

**Discussion by 2001herne** [davatorium/rofi#1601] [Ideas/Feature Requests]
Currently, Rofi's script mode ([rofi-script](https://github.com/davatorium/rofi/blob/next/doc/rofi-script.5.markdown)) allows the setting of an info variable for preset rows. This is useful for passing information back to the next iteration of the script. However, Rofi also allows users to enter custom entries that do not appear on the preset list. Currently, there does not appear to be a way to attach an info variable to these custom entries. My own personal thoughts are that this might get set along with the mode options (i.e.`echo -en "\0custom-info\x1fsome_info_here\n`), alongside the prompt, message text, etc.

**Comment by DaveDavenport** [comment]
What is the use-case for this?

**Reply by 2001herne** [reply]
My current use case is as a way of maintaining state between invocations of the script - currently, (in my use case) the `info` component of any row contains two pieces of data - a `verb` variable, instructing the script which code to execute on any given invocation, and a `args` component, supplying arguments to that section of code. This is trivial to do for preset rows where I can explicitly set the `info` component of the row but falls over when it comes to value entry - creating a file with a custom name, for instance.

**Reply by DaveDavenport** [reply]
So do you want to set 'data' for custom entry, or set some 'data' that is always exposed to the next execution?

**Reply by 2001herne** [reply]
Originally I thought the first, but now that I'm thinking further, the second would probably be more useful, as row-specific data and general state data could be better separated. I have noticed that I'm repeating a lot of the `info` between rows currently.

**Reply by DaveDavenport** [reply]
I think Ill add an option to pass data from one execution to the next via a header, sounds like a decent plan.

**Reply by 2001herne** [reply]
Thanks for the help.

**Reply by DaveDavenport** [reply]
implemented in git.

**Reply by 2001herne** [reply]
Just wanted to say it looks perfect. I'm gonna have some fun with this.
